bugbug/models/duplicate.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.pipeline import Pipeline
+import shap
+from bugbug import bug_features, bugzilla, feature_cleanup
+import pandas as pd
+from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer, TfidfVectorizer
+import scipy
+from sklearn.metrics import accuracy_score
+import numpy as np
+from sklearn.model_selection import train_test_split, cross_validate
+from sklearn.svm import LinearSVC
+
+
+
+class DuplicateModel:
+    def __init__(self, lemmatization=False):
+        self.cross_validation_enabled = False
+        self.sampler = None
+
+        self.calculate_importance = True
+        self.pipeline = Pipeline([
+            ("tfidf", TfidfVectorizer(preprocessor=self.transform)),
+            ("clf", LinearSVC())
+        ])
+
+    def transform(self, X):
+        cleanup_functions = [
+            feature_cleanup.responses(),
+            feature_cleanup.hex(),
+            feature_cleanup.dll(),
+            feature_cleanup.fileref(),
+            feature_cleanup.url(),
+            feature_cleanup.synonyms(),
+            feature_cleanup.crash(),
+        ]
+        for cleanup_function in cleanup_functions:
+            X = cleanup_function(X)
+        return X
+
+    def get_text(self, bug):
+        return bug["summary"] + " ".join(comment["text"] for comment in bug["comments"])
+    
+    def get_duplicate_map(self):
+        
+        bugs = bugzilla.get_bugs()
+        bug = {}
+        duplicate_bug_map = {}
+        count = 500
+        for bug_data in bugs:
+            bug_id = int(bug_data["id"])
+
+            if bug_data["resolution"] == "DUPLICATE":
```

Comment:
Instead of using the resolution, use the values of dupe_of and duplicates. This way you put in the map all bugs that are part of the duplicate group (both the bugs that are resolved as duplicate of a bug, and the bug which is the "main" one).